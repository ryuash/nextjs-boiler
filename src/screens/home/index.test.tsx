import React from 'react';
import { render, screen } from '@testing-library/react';
import Component from '.';

// ==================================
// mocks
// ==================================
const mockI18n = {
  t: (key: string) => key,
  lang: 'en',
};

jest.mock('next-translate/useTranslation', () => () => mockI18n);

// ==================================
// unit tests
// ==================================
describe('Dummy Test', () => {
  it('matches snapshot', async () => {
    const { asFragment } = render(<Component />);
    expect(screen.getByText(
      /Testing Next.js With Jest and React Testing Library/i,
    )).toBeInTheDocument();
    expect(screen.getByText(
      /hwohoh/i,
    )).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
