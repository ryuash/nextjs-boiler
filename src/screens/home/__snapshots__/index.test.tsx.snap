// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dummy Test matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Testing Next.js With Jest and React Testing Library
    </h1>
    <div>
      hello worsdfdsld
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
      tabindex="0"
      type="button"
    >
      <span
        class="MuiButton-label"
      >
        hwohoh
      </span>
      <span
        class="MuiTouchRipple-root"
      />
    </button>
    <p>
      one
    </p>
  </div>
</DocumentFragment>
`;
